List the development models at the top

Baalkikhaal opened the pull request:
Added an unordered list summarizing the collaborative development models.

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [contributing.md](/main/CONTRIBUTING.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:

<!-- 
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

The article does not start with a unordered list summarizing the content of the article as in the rest of the GitHub docs articles. Hence, for the purpose of symmetry, such a list is needed at the top of the article so reader gets an idea of what is the article.

### What's being changed:

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. -->

### Check off the following:
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (eg. a "before and after video")

<!-- Description of the writer impact here -->

Baalkikhaal committed:
List the development models at the top

Added an unordered list summarizing the collaborative development models.

janiceilene commented on content/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models.md:
```diff
@@ -12,7 +12,12 @@ topics:
-There are two main types of development models with which you'd use pull requests. In the *fork and pull model*, anyone can fork an existing repository and push changes to their personal fork. You do not need permission to the source repository to push to a user-owned fork. The changes can be pulled into the source repository by the project maintainer. When you open a pull request proposing chang ...
+There are two main types of collaborative development models with which you'd use pull requests.
+
+- Fork and pull model
```
Thank so much for opening a PR @Baalkikhaal! You're right, this page is woefully out of date and could use some attention 💛 

Maybe instead of adding them as an unordered list manually, we can update the intro (in the front matter) to read:

> The way you use pull requests depends on the type of development model you use in your project. You can use the fork and pull model or the shared repository model.

Then you can remove this unordered list and create subheaders for the two sections instead. With subheadings, the table of contents will be autogenerated, just like it is for the rest of the docs.

While you're here, would you mind removing the italics around *fork and pull* and *shared repository model*? We don't use italics in our docs anymore.

Would you ping me once you've made these changes and I'll get this up for review?

Baalkikhaal replied:
Thank you @janiceilene for replying to the pull request. Actually I am trying to wade through the waters of pull-requests cautiously. Honestly, I need some hand-holding from people like you😅. I will ping you once I understand how to update your recommended changes to the PR 👍?

Baalkikhaal replied:
@janiceilene I have applied the changes that you have suggested. Please have a look into it 😄.

Baalkikhaal committed:
1. Update frontmatter 2. Add subheadings

Intro property of front matter has been updated to reflect contents of the article. Subheadings are added to the article to improve the logical structure of article.

janiceilene reviewed (approved):
I checked out your changes on staging (https://docs-5759--patch-1.herokuapp.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models) and they look great! I'll update this branch and get this merged once tests are passing. Thank you so much for opening a PR for the fix and then taking the time to make even more changes to this out of date article 🎉

janiceilene committed:
Merge branch 'main' into patch-1

Baalkikhaal commented:
Thank you. This is my first Pull Request ever on any open source project 😄 .

> I checked out your changes on staging (https://docs-5759--patch-1.herokuapp.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models) and they look

This link does not open for me. Do I need to install Heroku?

janiceilene commented:
Congrats on your first open source PR 🎉 🎉 🎉 

> This link does not open for me. Do I need to install Heroku?

The link only works while the PR is still open. Once it's closed or merged, the changes can be viewed on docs.github.com and the staging server link doesn't work anymore. So your changes are visible on https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models 💖
